### ISSUE TITLE
loss_mask the opposite of what I expected

### ISSUE THREAD

**Issue by fhuszar** [google-deepmind/neural_networks_solomonoff_induction#4]
I tried sampling from the data generator, using the following lines of code (mostly copied from `train.py`):

```python
from data import utm_data_generator as utm_dg
from data import utms

utm = utms.BrainPhoqueUTM()
data_generator = utm_dg.UTMDataGenerator(
      batch_size=32,
      seq_length=256,
      rng=1,
      utm=utm,
      memory_size=10,
      maximum_steps=100,
      tokenizer=utm_dg.Tokenizer.ASCII,
      maximum_program_length=100,
  )
batch, log_dict = data_generator.sample()
loss_mask = log_dict['loss_mask']
print(loss_mask)
```

I obtain a  mask that has 0s at the beginning of each datapoint, and 1s at the end. This is consistent with the description of the `mask`  in `_make_loss_fn`. However, in the implementation, the mask is used like so:
```python
    true_conditionals = jnp.multiply(mask, true_conditionals)
    marginals = jnp.sum(true_conditionals, axis=1)  # Shape (B,).
```
To me, this would make sense if the 1s in the mask denoted the conditionals we want to keep, but instead we have 1s in positions we want to ignore, as documented.

Is this a possible bug? Or please clarify if I misunderstood how the masks were used.

**Comment by jordigraumo** [comment]
Yes, there seems to be a slight mismatch between our internal code and this. We will fix it as soon as we can. In essence, you just need to add `mask = 1 - mask` before `true_conditionals`, but I am going to check that there is nothing else going on. Thanks for flagging!

**Comment by anianruoss** [comment]
Indeed, we use `jnp.where(mask, 0.0, true_conditionals)` internally instead of `jnp.multiply(mask, true_conditionals)`. I sent a fix for review and will update the code here soon.

Thanks for flagging!
